ast/compare.go

```diff
@@ -83,7 +82,15 @@ func Compare(a, b interface{}) int {
 		}
 		return 1
 	case Number:
-		return util.Compare(json.Number(a), json.Number(b.(Number)))
+		bigA, ok := new(big.Float).SetString(string(a))
+		if !ok {
+			panic("illegal value")
+		}
+		bigB, ok := new(big.Float).SetString(string(json.Number(b.(Number))))
```

Comment:
Out of curiosity (I don't really have much context on what `b` is yet), why does it need to go through the `json.Number` stringer? Does `string(b.(Number))` not work as the input to the big float `SetString`?

Reply:
@patrick-east that's a good point, I think you can just go from ast.Number->string directly.

Reply:
Oops, that was me not thinking!